xs2a-adapter-service-api-adapter/src/main/java/de/adorsys/xs2a/adapter/http/ApacheHttpClient.java

```diff
@@ -120,12 +130,49 @@ public String content(Request.Builder requestBuilder) {
             InputStream content = entity != null ? entity.getContent() : EmptyInputStream.INSTANCE;
 
             T responseBody = responseHandler.apply(statusCode, content, responseHeaders);
+            logResponse(response, responseHeadersMap, responseBody);
             return new Response<>(statusCode, responseBody, responseHeaders);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    private <T> void logResponse(
+        CloseableHttpResponse response,
+        Map<String, String> headers,
+        T responseBody
+    ) {
+        logger.debug("<-- {}", response.getStatusLine());
+        headers.keySet().forEach(
+            key -> logger.debug("<-- {}: {}", key, headers.get(key))
+        );
+        logger.debug("<--");
+        Optional<String> contentType = Optional.ofNullable(
```

Comment:
Please use a normal null check. We went over this a dozen times now. Why are you keep doing this?!

Reply:
replaced